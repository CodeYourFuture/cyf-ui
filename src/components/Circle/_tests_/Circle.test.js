import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { CircleWithNum } from "..";

describe("<CircleWithNum/>", () => {
  const num = 0;
  const mockOnClick = jest.fn(() => null);
  test("Match the snapShot", () => {
    const { container } = render(
      <CircleWithNum num={num} onClick={mockOnClick} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test("calls onClick when it is clicked", () => {
    render(<CircleWithNum num={num} onClick={mockOnClick} />);
    const circle = screen.getByTestId("circle");
    fireEvent.click(circle);
    expect(mockOnClick).toHaveBeenCalled();
  });
});
