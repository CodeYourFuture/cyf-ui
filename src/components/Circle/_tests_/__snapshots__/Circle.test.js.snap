// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CircleWithNum/> Match the snapShot 1`] = `
.c0 {
  border-radius: 50%;
  width: 16px;
  height: 16px;
  padding: 3px;
  background: #335eff;
  border: 2px solid #335eff;
  color: white;
  text-align: center;
  font-weight: 800;
  font: 13px Arial,sans-serif;
}

<div
  class="c0"
  data-testid="circle"
>
  0
</div>
`;
